### PR TITLE
PMM-4676 Adding changes for custom branch support to trigger tests 

### PR DESCRIPTION
**Pull request by puneet0191**

### CODE REVIEW

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-652-da777a9
client docker - perconalab/pmm-client-fb:PR-652-da777a9
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-652-da777a9.tar.gz

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-652-b6605fa
client docker - perconalab/pmm-client-fb:PR-652-b6605fa
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-652-b6605fa.tar.gz

File: .gitmodules [hunk 1]
```diff
@@ -82,3 +82,13 @@
...
 	url = https://github.com/Percona-Lab/percona-images
 	branch = master
+
+## PMM Tests
+[submodule "pmm-api-tests"]
+	path = sources/pmm-api-tests/src/github.com/Percona-Lab/pmm-api-tests
+	url = https://github.com/Percona-Lab/pmm-api-tests
+	branch = master
+[submodule "pmm-qa"]
+	path = sources/pmm-qa/src/github.com/percona/pmm-qa
```
Comment (AlekSi) [inline .gitmodules]: We need those long paths for Go code (for GOPATH). Do we need it there? Can we simplify it to `sources/pmm-qa`?

File: Jenkinsfile [hunk 2]
```diff
@@ -3,10 +3,12 @@ library changelog: false, identifier: 'lib@master', retriever: modernSCM([
     remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
 ]) _
 
-void runAPItests(String DOCKER_IMAGE_VERSION, CLIENT_VERSION, OWNER) {
+void runAPItests(String DOCKER_IMAGE_VERSION, BRANCH_NAME, API_COMMIT_SHA, CLIENT_VERSION, OWNER) {
     stagingJob = build job: 'pmm2-api-tests', parameters: [
         string(name: 'DOCKER_VERSION', value: DOCKER_IMAGE_VERSION),
+        string(name: 'GIT_BRANCH', value: BRANCH_NAME),
         string(name: 'OWNER', value: OWNER),
+        string(name: 'GIT_COMMIT_HASH', value: API_COMMIT_SHA)
```
Comment (AlekSi) [inline Jenkinsfile]: I always wondered why we use so different strings for names and values. Isn't it confusing? Can we use `GIT_COMMIT_HASH` for both, for example?

File: Jenkinsfile [hunk 3]
```diff
@@ -61,8 +64,19 @@ pipeline {
                     git lfs pull
                     git lfs checkout
                     cd $curdir
+                    export api_commit_sha=$(git submodule status | grep 'pmm-api-tests' | awk -F ' ' '{print $1}')
+                    export api_branch=$(git config -f .gitmodules submodule.pmm-api-tests.branch)
```
Comment (AlekSi) [inline Jenkinsfile]: Our APIs are defined in `pmm` repo. Let's use a longer name for API tests. Like `apt_tests_branch`, etc.

File: .gitmodules (hunk 1 above)
Comment (puneet0191) [reply .gitmodules]: I did that exactly during the initial commit and you requested to change to GOPATH :confused: so for this iteration, we can keep it same, I will merge change in next PR with test suite

Comment (AlekSi) [reply .gitmodules]: We need GOPATH for github.com/Percona-Lab/pmm-api-tests above. I think I requested it for that repo.

**Comment by JNKPercona** [comment]
server docker - perconalab/pmm-server-fb:PR-652-8243f5e
client docker - perconalab/pmm-client-fb:PR-652-8243f5e
client - https://s3.us-east-2.amazonaws.com/pmm-build-cache/PR-BUILDS/pmm2-client/pmm2-client-PR-652-8243f5e.tar.gz

**Pull request merged**